### PR TITLE
Added treedata conversion skeleton

### PR DESCRIPTION
**Pull request by isabellechan089**
Code to convert Cassiopeia Tree to treedata

### CODE REVIEW

File: cassiopeia/data/utilities.py [hunk 1]
```diff
@@ -681,3 +684,128 @@ def compute_inter_cluster_distances(
             inter_cluster_distances.loc[state1, state2] = distance
 
     return inter_cluster_distances
+
+def cassiopeia_to_treedata(
+    cassiopeia_tree,
+    tree_key: str = "lineage",
+    preserve_layers: bool = True,
+    preserve_metadata: bool = True,
+    alignment: str = "leaves"
```
[the remark this answers is not in the document]
Comment (isabellechan089) [reply cassiopeia/data/utilities.py:693]: Got it, I believe I mixed this up with treedata's allowed alignments.

File: cassiopeia/data/utilities.py [hunk 2]
```diff
@@ -681,3 +684,128 @@ def compute_inter_cluster_distances(
...
+        - X = None
+        - obsm["character_matrix"] = character matrix
+        - obst[tree_key] = tree topology
+    """
+    
+    # X is None - typically RNA data which CassiopeiaTree doesn't have
+    X = None
+    
+    # Validate that we have both character matrix and tree
+    if cassiopeia_tree.character_matrix is None:
```
[the remark this answers is not in the document]
Comment (isabellechan089) [reply cassiopeia/data/utilities.py:722]: Last meeting, we discussed requiring a character matrix/tree in the converter function (under the assumption that it didn't make much sense to have this object w/o a character matrix or tree). However, I can go ahead and remove this assumption if we are supporting these cases!

[the remark this answers is not in the document]
Comment (colganwi) [reply cassiopeia/data/utilities.py:722]: I generally favor making software more flexible since there may be a use case you haven't thought of. Someone could simulate the tree topology using Cassiopeia but not the character matrix and then want to convert it to TreeData.

**Review by colganwi — CHANGES_REQUESTED** [review]
Thanks @isabellechan089! I just have a few minor comments. Once the CassiopeiaTree.to_treedata() method is implemented this should be good to go.

There are a few edge cases you should add to your tests to make sure they work. For example, what if there is no tree or no character matrix? You should test that the dissimilarity map is copied.
